### PR TITLE
[CodeQuality] Handle Nested Ternary on SwitchNegatedTernaryRector to have parenthesis

### PR DESCRIPTION
**Pull request by samsonasik**
Having code: 

```php
return ! isset($result['validation_messages'])
            ? []
            : (isset($result['validation_messages']['http']) ? $result['validation_messages'] : $result);
```

It currently generate:

```diff
return isset($result['validation_messages'])
+            ? isset($result['validation_messages']['http']) ? $result['validation_messages'] : $result
+            : ([]);
```

which should be:

```diff
return isset($result['validation_messages'])
+            ? (isset($result['validation_messages']['http']) ? $result['validation_messages'] : $result)
+            : [];
```

### CODE REVIEW

**Comment by samsonasik** [comment]
@TomasVotruba is there an available method to handle wrap / unwrap parenthesis in nested ternary expr?

**Comment by TomasVotruba** [comment]
Nothing I know of :thinking: 
It might be useful to extract this logic to a standalone service for future use. E.g. `TernaryBracketWrapper`

**Comment by samsonasik** [comment]
Fixed 🎉 with create `new Name()` with Print https://github.com/rectorphp/rector-src/pull/397/commits/c3a7f8399c9957ca240d7c519e27a65923a285e7 . 

To remove left over `()` from non-ternary after flip, I think that's should be handled by coding standard, like using `no_unneeded_control_parentheses` for php-cs-fixer.

**Comment by samsonasik** [comment]
Moved the logic to `TernaryBracketWrapper` service.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: src/PhpParser/Node/Value/TernaryBracketWrapper.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public function __construct(private BetterStandardPrinter $betterStandardPrinter)
+    {
+    }
+
+    public function getWrappedTernaryConstFetch(Ternary $ternary): ConstFetch
+    {
+        return new ConstFetch(new Name('(' . $this->betterStandardPrinter->print($ternary) . ')'));
+    }
+}
```
Comment (TomasVotruba) [inline src/PhpParser/Node/Value/TernaryBracketWrapper.php]: This is rather hacking node contents. The `(` `)` actually are not part of the `Name`.

More clean way is to use attribute and then extend `BetterStandardPrinter` to print this node with the attribute correctly. E.g. `Array_` has `setAttribute('kind', ...)` value that handles 2 prints:

```php
array()
[]
```

Here in this file that means something like this:

```php
$ternary->setAttribute('kind', 'wrapped_with_brackets');
```

Then add a new overload method to `BetterStandardPrinter`:

```php
protected function pExpr_Ternary(Expr\Ternary $node)
{
    // see the parent method for inspiration
    // ...
}
```

Comment (samsonasik) [reply src/PhpParser/Node/Value/TernaryBracketWrapper.php]: I will try.

Comment (samsonasik) [reply src/PhpParser/Node/Value/TernaryBracketWrapper.php]: implemented https://github.com/rectorphp/rector-src/pull/397/commits/92d341ae4778610f7f0b93295881065c4eb99d9d#diff-35b384f5686672e9b726b991a0c45dd8c18cfc8b365953489bf95a39db82205fR367-R375

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Neat job, thank you

**Pull request merged**